src/app/games/game/game-progress/game-chart/game-chart.component.ts

```diff
@@ -0,0 +1,143 @@
+import { Component, Input, OnChanges, OnInit } from '@angular/core';
+import { Chart } from 'angular-highcharts';
+import { Options, PointOptionsObject } from 'highcharts';
+import { IChart } from 'src/app/games/games.service.interface';
+
+/**
+ * The container for a HighCharts chart.
+ */
+@Component({
+  selector: 'game-chart',
+  templateUrl: './game-chart.component.html',
+  styleUrls: ['./game-chart.component.scss']
+})
+export class GameChartComponent implements OnChanges {
+
+  constructor() { }
+
+  /**
+   * The matched CSV data to use for this chart.
+   */
+  @Input() matched: string = null;
+  /**
+   * The unmatched CSV data to use for this chart.
+   */
+  @Input() unmatched: string = null;
+  /**
+   * The chart metadata stored in the game settings.
+   */
+  @Input() metadata: IChart = null;
+  /**
+   * The HighCharts chart instance.
+   */
+  chart: Chart = null;
+
+  ngOnChanges(): void {
+    // don't do anything until all inputs are provided
+    if (typeof this.matched !== "string" || typeof this.unmatched !== "string" || typeof this.metadata !== "object") {
+      return;
+    }
+
+    // set up chart data
+    const metadata = this.metadata;
+    let matchedData = this.parseData(this.matched);
+    let unmatchedData = this.parseData(this.unmatched);
+
+    let joined = Array.prototype.concat(matchedData, unmatchedData);
+    joined.sort((a, b) => a["y"] - b["y"]);
+    const interval = 1 / 5;
+    const max = Math.max(interval, Math.ceil(joined.slice(-1)[0].y / interval) * interval);
+
+    const options: Options = {
+      chart: { type: "line" },
+      title: { text: this.metadata.title },
+      subtitle: { text: this.metadata.subtitle },
+      tooltip: {
+        formatter: function() {
+          const data = this.point.options;
+          let tooltip = "";
+
+          tooltip += `Date: ${new Date(data.x).toLocaleString()}<br/>\n`;
+          tooltip += `Commit: ${data["commit"]}<br/>\n`
+          tooltip += `Total ${metadata.series[0].metric}: ${(+data.y * 100).toFixed(2)}%<br/>\n`
+          tooltip += `-------------------------------------------<br/>\n`;
+
+          for (let i = 0; i < metadata.series.length - 1; i += 1) {
+            const text = metadata.series[i + 1].description
+              .replace("{0}", data["metrics"][i][0])
+              .replace("{1}", data["metrics"][i][1])
+              .replace("{2}", (+data["metrics"][i][2] * 100).toFixed(2));
+            tooltip += `${text}<br/>\n`;
+          }
+
+          return tooltip;
+        }
+      },
+      xAxis: {
+        title: { text: "Date" },
+        type: "datetime"
+      },
+      yAxis: {
+        title: { text: "Completion (%)" },
```

Comment:
```suggestion
        title: { text: "Matched (%)" },
```
Since the objective is matching would it make sense to have the axis labeled for matched instead of completion?

Reply:
I wouldn't have thought this made sense as we have "matched" and "non-matched" datasets. Thoughts?

Reply:
Maybe "Code decompiled".

Reply:
I would suggest just "decompiled", as it has to apply to stuff that isn't code, like assets

Reply:
Although aren't the objectives a mix of decompilation and matching? So completion might be best after all?

Reply:
Well, it doesn't really make sense to "decompile" assets: the technical term is "debinarisation".

Reply:
Maybe just "Completion" or something generic then, if it's intended to be generic.

Reply:
Exactly, so the fact that there are many different tasks would lead me to believe that a generic "Completion" is likely the best option

Reply:
Any further thoughts on this @jpburnett?